### PR TITLE
rust 1.0.0

### PR DESCRIPTION
**Pull request by sferik**
http://blog.rust-lang.org/2015/05/15/Rust-1.0.html

In addition to updating the `rustc` source tarball, it also updates `rustc-nightly` and `cargo-nightly`.

Does anything need to be done to update the bottles or does the @BrewTestBot do that automatically?

### CODE REVIEW

**Comment by mikemcquaid** [comment]
> Does anything need to be done to update the bottles or does the @BrewTestBot do that automatically?

It'll handle them.

**Comment by sferik** [comment]
> It'll handle them.

@mikemcquaid Thanks!

**Comment by sferik** [comment]
Note: This pull request also reverts the `--release-channel=beta` flag, which was added in 9721953dd5ff3da7ed08809d22831323c9d39efd.

I believe this brings the Homebrew formula back in line with the official builds, but I may be wrong about that.

**Comment by jonathanmarvens** [comment]
:+1::+1::+1:

File: Library/Formula/rust.rb [hunk 1]
```diff
@@ -44,7 +44,6 @@ def install
     args = ["--prefix=#{prefix}"]
     args << "--disable-rpath" if build.head?
     args << "--enable-clang" if ENV.compiler == :clang
-    args << "--release-channel=beta" unless build.head?
```
Comment (steveklabnik) [inline Library/Formula/rust.rb]: you may or may not want to re-add a `--release-channel=nightly" if build.head?` too

Comment (kballard) [inline Library/Formula/rust.rb]: Should this not say `--release-channel=stable`? It looks like the default channel is "dev" even in the 1.0.0 tarball.

Comment (kballard) [inline Library/Formula/rust.rb]: @steveklabnik Is `nightly` correct for downloading from `HEAD`? I would think `dev` is fine there.

Comment (callahad) [inline Library/Formula/rust.rb]: From reading the current [configure](https://github.com/rust-lang/rust/blob/e5394240a295650b567aa406b4a0e1e3a6749a5f/configure#L645-L656) and (make.mk)[https://github.com/rust-lang/rust/blob/e5394240a295650b567aa406b4a0e1e3a6749a5f/mk/main.mk#L27-L54] scripts, it looks like `nightly` is correct for HEAD, and we need `stable` when building 1.0

File: Library/Formula/rust.rb [hunk 2]
```diff
@@ -1,25 +1,25 @@
 class Rust < Formula
   homepage 'http://www.rust-lang.org/'
-  version "1.0.0-beta.4"
+  version "1.0.0"
```
Comment (bfontaine) [inline Library/Formula/rust.rb]: You can leave out this line because Homebrew infers the version from the URL.

**Comment by mikemcquaid** [comment]
Install is failing on all platforms, unfortunately: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/26224/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_rust/

**Comment by sferik** [comment]
@mikemcquaid I just pushed an update. `cargo` was locked to the wrong revision. Now `cargo` is locked to the same revision as the Rust 1.0.0 installer.

**Comment by sferik** [comment]
@mikemcquaid Ping. Tests are now passing. Should be good to merge!

**Comment by antstorm** [comment]
:+1:

**Comment by byronyi** [comment]
When I run rustc, I got the following error information:
```
$ rustc
dyld: Library not loaded: x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_driver-4e7c5e5c.dylib
  Referenced from: /usr/local/bin/rustc
  Reason: image not found
Trace/BPT trap: 5
```

**Comment by kballard** [comment]
By any chance are you running it from within the root dir of a checkout of the [rust-lang/rust](https://github.com/rust-lang/rust) repo? My recollection is that due to various reasons, attempting to run `rustc` from that folder does not work properly as it attempts to load dylibs from the stage paths instead of from `/usr/local/lib`.

FWIW, configuring rust with `--enable-rpath` (which makes it use `@rpath` when linking rustc itself) seems to fix this issue. I'm not sure offhand why rpaths are disabled by default (and when using `rustc` to build stuff yourself, it won't use rpaths for the built product unless you pass `-C rpath` as a compiler flag to `rustc`).

**Comment by byronyi** [comment]
```
$ rustc --version
dyld: Library not loaded: x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_driver-4e7c5e5c.dylib
  Referenced from: /usr/local/bin/rustc
  Reason: image not found
Trace/BPT trap: 5
```

**Comment by l-hedgehog** [comment]
According to the original plan (#38674), I think the 2015-04-30 nightly of cargo should be used with rust 1.0 rust-lang/rust-packaging@37686ecb5465, instead of the current 2015-05-14 one?

**Comment by sferik** [comment]
@l-hedgehog My original pull request had `cargo` at revision `efb482d9ed820e26e79329d73bdd0a26c4cebf41`, which is the same version that ships with the Rust Mac installer. Unfortunately, this version of `cargo` caused the build to fail with Rust 1.0.0, so I updated to the latest revision at the time of the 1.0.0 release. You can see the console output of that failed build [here](http://bot.brew.sh/job/Homebrew%20Pull%20Requests/26242/version=yosemite/console). In the process of compiling `term`, a dependency of `tar`, it fails with the following error:

```
/Users/brew/Jenkins/prs/version/yosemite/home/.cargo/registry/src/github.com-1ecc6299db9ec823/tar-0.2.10/src/lib.rs:186:27: 186:31 error: mismatched types:
 expected `u16`,
    found `i32`
```

If you can find a way to use this version of `cargo` without causing the build to fail, please open a pull request.

**Comment by l-hedgehog** [comment]
@sferik thanks for the detailed explanation

**Pull request closed without merging**